File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidconnectionsconnectioniddatacommerceordersorderid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class GetCompaniesCompanyIdConnectionsConnectionIdDataCommerceOrdersOrderIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=connectionId" })
  connectionId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=orderId" })
  orderId: string;
}

export class GetCompaniesCompanyIdConnectionsConnectionIdDataCommerceOrdersOrderIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdConnectionsConnectionIdDataCommerceOrdersOrderIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdConnectionsConnectionIdDataCommerceOrdersOrderIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdConnectionsConnectionIdDataCommerceOrdersOrderIdSecurity;
}

export class GetCompaniesCompanyIdConnectionsConnectionIdDataCommerceOrdersOrderIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsCommerceOrder?: shared.CodatDataContractsDatasetsCommerceOrder;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getcompaniescompanyidconnectionsconnectioniddatacommerceordersorderid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDConnectionsConnectionIDDataCommerceOrdersOrderIDPathParams struct {
	CompanyID    string `pathParam:"style=simple,explode=false,name=companyId"`
	ConnectionID string `pathParam:"style=simple,explode=false,name=connectionId"`
	OrderID      string `pathParam:"style=simple,explode=false,name=orderId"`
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataCommerceOrdersOrderIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataCommerceOrdersOrderIDRequest struct {
	PathParams GetCompaniesCompanyIDConnectionsConnectionIDDataCommerceOrdersOrderIDPathParams
	Security   GetCompaniesCompanyIDConnectionsConnectionIDDataCommerceOrdersOrderIDSecurity
}

type GetCompaniesCompanyIDConnectionsConnectionIDDataCommerceOrdersOrderIDResponse struct {
	CodatDataContractsDatasetsCommerceOrder *shared.CodatDataContractsDatasetsCommerceOrder
	ContentType                             string
	StatusCode                              int
}
